src/modules/chat/ChatModuleNew.tsx

```diff
@@ -22,10 +26,75 @@ export const ChatModuleNew = () => {
     })()
   }, [account, library]);
 
+ useEffect(() => {
+    if(!connectedUser) return;
+    (async function () {
+      try {
+        const inboxes = await getInboxFromIndexDb();
+        fetchInbox();
+        const requests = await getRequestsFromIndexDb();
+        fetchRequests();
+        setInbox(inboxes,account);
+        setRequests(requests,account);
+      } catch (err) {
+        console.log(err);
+      }
+    })()
+  }, [connectedUser]);
+
+  const getInboxFromIndexDb = async ():Promise<FeedsNew> => {
+    //remove old indexDb structure
+      const inboxesFromIndexDb:any = await intitializeDb<string>('Read', 'Inbox', walletToCAIP10({ account }), '', 'did');
+
+      if (inboxesFromIndexDb !== undefined) {
+        let inboxes: FeedsNew = inboxesFromIndexDb.body;
+        //change decryptFeeds to a common function
+        const decryptedInboxes: FeedsNew = await decryptFeedsNew({ feeds: inboxes, connectedUser });
+        console.log(decryptedInboxes)
+         return decryptedInboxes;
+      }
+  };
+
+  const fetchInbox = async ():Promise<void> => {
+    const inboxes = await getInbox(account);
+    if(inboxes.length)
+    {
+      const mappedInboxes:FeedsNew = Object.assign({}, ...inboxes.map(s => ({[getChatId(s)]: s})));
+      intitializeDb<FeedsNew>('Insert', 'Inbox', walletToCAIP10({ account: connectedUser.wallets! }), mappedInboxes, 'did');
+      const decryptedInboxes:FeedsNew = await decryptFeedsNew({ feeds: mappedInboxes, connectedUser });
+      console.log(decryptedInboxes)
+      setInbox(decryptedInboxes,account);
+    }
+  }
+
+  const getRequestsFromIndexDb = async ():Promise<FeedsNew> => {
```

Comment:
this one too

Reply:
in fact, create new file in helpers folders, indexedDB.ts and add all indexedDB helper functions in that